ballerina/tests/client_initialization_test.bal

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2025 WSO2 LLC (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+const string VALID_MCP_URL = "http://localhost:3000/mcp";
+const string UNREACHABLE_MCP_URL = "http://localhost:9999/mcp";
+const string INVALID_URL = "invalid-url";
+const string TEST_CLIENT_NAME = "test-client";
+const string TEST_CLIENT_NAME_2 = "test-client-2";
+const string TEST_CLIENT_VERSION = "1.0.0";
+
+@test:Config {}
+function testClientConstructionWithValidUrl() returns error? {
+    StreamableHttpClient 'client = check new (VALID_MCP_URL);
```

Comment:
Can't we specify the client variable in the module level instead of initializing in each test case

Reply:
we can't do this, as we are testing multiple ways of calling `initialize()` on a client and we can call `initialize()` once on a client, that's why we have to initialize the client on each test function.

Reply:
Actually these tests are running on a stateless server, in that case, calling initialization() will not gonna create a state. so we can call multiple initialze call with the same behaviour. With that I addressed this. to have a global declaration.